pkg/api/controller.go

```diff
@@ -1023,12 +1023,15 @@ func (c *Controller) LinkPhysicalAddress(w http.ResponseWriter, r *http.Request,
 
 // normalizePhysicalAddress return relative address based on storage namespace if possible. If address doesn't match
 // the storage namespace prefix, the return address type is full.
-func normalizePhysicalAddress(storageNamespace, physicalAddress string) (string, catalog.AddressType) {
+func (c *Controller) normalizePhysicalAddress(storageNamespace, physicalAddress string) (string, catalog.AddressType) {
 	prefix := storageNamespace
 	if !strings.HasSuffix(prefix, catalog.DefaultPathDelimiter) {
 		prefix += catalog.DefaultPathDelimiter
 	}
-	if strings.HasPrefix(physicalAddress, prefix) {
+	dataPrefix := prefix + c.PathProvider.GetDataPathPrefix() + catalog.DefaultPathDelimiter
```

Comment:
GetDataPathPrefix can return empty string, we should consider it and append catalog.DefaultPathDelimiter only if GetDataPathPrefix didn't ended with prefix